### ISSUE TITLE
ListBoxWithTags - a11y defects

### ISSUE THREAD

**Issue by mikrotron** [acl-services/paprika#820]
Since I haven't actually evaluated the existing `<ListBox>` component, I think a number of these actually apply more generally to it, rather than the new `WithTags` version.  

Also, since I just tested the story in Storybook, it's possible some might relate the the example implementation instead of the actual component.  But since those stories are inspiration for consumers, they should probably be fixed anyway. 

1. Most of our buttons can be activated by `space` but the trigger for the `<ListBoxWithTags>` does not open the list.
2. The trigger button should have `aria-haspopup="listboxId_"` and `aria-expanded` attributes.
3. For the list of selected items (tags) in the trigger, there's a couple `<div>` elements between the `<ul>` and `<li>`.  This is [invalid markup])(https://validator.w3.org/nu/#textarea) – I suggest you use aria roles instead (`list` / `listitem`) which will allow `div` elements in between.
~4. The search box in the drop down seems to trap the keyboard focus (pressing `tab` does not move focus – even though `shift` + `tab` _does_ move it backwards to the trigger).~ 
5. When using the `↓` key to move the selection highlight in the list, nothing is announced by the screenreader (probably since actual document focus remains on the search box).

#### References
https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html
https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html

**Comment by nahumzs** [comment]
4. Just occurs in a situation where there is nothing to focus on apart of the WithTags component which is unlikely to ocurrs.

**Comment by oscarkwan** [comment]
3. Can't reproduce

**Comment by mikrotron** [comment]
Cool, `3` was inadvertently resolved in Tristan's PR to extract the `<Tag>`
